### PR TITLE
Fix CI s3 api command to fetch latest results

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description
An `s3` fetch api seems to be returning incorrectly ordered output, this PR fixes the issue.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: ci/cudf_pandas_scripts/pandas-tests/diff.sh [hunk 1]
```diff
@@ -17,10 +17,9 @@ MAIN_ARTIFACT=$(rapids-s3-path)cuda12_$(arch)_py${PY_VER}.main-${RAPIDS_FULL_VER
 PR_ARTIFACT=$(rapids-s3-path)cuda12_$(arch)_py${PY_VER}.pr-${RAPIDS_FULL_VERSION}-results.json
 
 rapids-logger "Fetching latest available results from nightly"
-aws s3api list-objects-v2 --bucket rapids-downloads --prefix "nightly/" --query "sort_by(Contents[?ends_with(Key, '_py${PY_VER}.main-${RAPIDS_FULL_VERSION}-results.json')], &LastModified)[::-1].[Key]" --output text > s3_output.txt
-
-read -r COMPARE_ENV < s3_output.txt
-export COMPARE_ENV
+aws s3api list-objects-v2 --bucket rapids-downloads --prefix "nightly/" --query "sort_by(Contents[?ends_with(Key, '_py${PY_VER}.main-${RAPIDS_FULL_VERSION}-results.json')], &LastModified)[::].[Key]" --output text > s3_output.txt
+cat s3_output.txt
```
Comment (galipremsagar) [inline ci/cudf_pandas_scripts/pandas-tests/diff.sh:21]: Leaving this for debug purposes to see what all files are being fetched.

Comment (raydouglass) [inline ci/cudf_pandas_scripts/pandas-tests/diff.sh:21]:
```suggestion
aws s3api list-objects-v2 --bucket rapids-downloads --prefix "nightly/" --query "sort_by(Contents[?ends_with(Key, '_py${PY_VER}.main-${RAPIDS_FULL_VERSION}-results.json')], &LastModified)[::].[Key]" --output text  | tee s3_output.txt
```

**Review by raydouglass — APPROVED** [review]
Just a small suggestion, but looks good.

**Pull request merged**